flake.nix

```diff
@@ -107,23 +107,38 @@
                                     then pkgs.pkgsCross.aarch64-multiplatform-musl
                                     else pkgs.pkgsCross.musl64
                                else pkgs;
-             in (__mapAttrs (compiler-nix-name: compiler:
+             in (builtins.mapAttrs (compiler-nix-name: compiler:
                   import ./dynamic.nix { inherit pkgs compiler compiler-nix-name; }
                   ) (compilers pkgs)
               // pkgs.lib.mapAttrs' (compiler-nix-name: compiler:
                   pkgs.lib.nameValuePair "${compiler-nix-name}-minimal" (
                     import ./dynamic.nix { inherit pkgs compiler compiler-nix-name; withHLS = false; withHlint = false; }
                   )) (compilers pkgs)
-              // pkgs.lib.mapAttrs' (compiler-nix-name: compiler: 
+              // pkgs.lib.mapAttrs' (compiler-nix-name: compiler:
                   pkgs.lib.nameValuePair "${compiler-nix-name}-static" (
                     import ./static.nix { pkgs = static-pkgs; inherit compiler compiler-nix-name; }
                   )) (compilers static-pkgs.buildPackages)
-              // pkgs.lib.mapAttrs' (compiler-nix-name: compiler: 
+              // pkgs.lib.mapAttrs' (compiler-nix-name: compiler:
                   pkgs.lib.nameValuePair "${compiler-nix-name}-static-minimal" (
                     import ./static.nix { pkgs = static-pkgs; inherit compiler compiler-nix-name; withHLS = false; withHlint = false; }
                   )) (compilers static-pkgs.buildPackages)
              );
-        hydraJobs = devShells;
+        hydraJobs = devShells // (pkgs.lib.mapAttrs' (name: drv:
+          pkgs.lib.nameValuePair "${name}-closure"
+          (pkgs.runCommand "${name}-closure" {
+            requiredSystemFeatures = [ "recursive-nix" ];
+            nativeBuildInputs = [ pkgs.nix pkgs.zstd pkgs.zip ];
+          } ''
+            mkdir -p $out/nix-support
+            HOME=$(mktemp -d)
+            nix-store --export $(nix-store -qR ${drv}) | zip -9 > $out/closure.zip
+            echo "file binary-dist \"$out/closure.zstd\"" >> $out/nix-support/hydra-build-products
```

Comment:
This does not line up. We generate a .zip file, but put a .zstd into the hydra-build-products?